cdap-ui/app/directives/complex-schema/array-schema/array-schema.html

```diff
@@ -0,0 +1,38 @@
+<!--
+  Copyright © 2016 Cask Data, Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+  use this file except in compliance with the License. You may obtain a copy of
+  the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+  License for the specific language governing permissions and limitations under
+  the License.
+-->
+
+<div class="array-schema-container">
+  <div class="clearfix">
+    <div class="input-column no-type">
+      <select class="form-control"
+              ng-model="ArraySchema.items.displayType"
+              ng-options="option as option for option in ArraySchema.SCHEMA_TYPES"
+              ng-change="ArraySchema.items.type = ArraySchema.items.displayType">
```

Comment:
We can't use onChange at the moment, because the digest cycle happening at different time, so the nested directive do not get the correct type during initialize stage. We might be able to optimize this after the next iteration of solving the current nested schema limit.